### DISCUSSION TITLE
EmoLLM-心理健康大模型

### DISCUSSION THREAD

**Discussion by aJupyter** [InternLM/Tutorial#209] [project]
# EmoLLM-心理健康大模型
## 👋 Welcome!
*仓库地址：*  https://github.com/aJupyter/EmoLLM
*OPenXLab EmoLLM：*  https://openxlab.org.cn/apps/detail/jujimeizuo/EmoLLM 
*视频链接：* https://www.bilibili.com/video/BV1N7421N76X/

最近规划：基于InternLM3打造RL+agent+更新新的模型+CT
---

 ## 介绍：
心理健康大模型（Mental Health Grand Model）是一个综合性的概念，它旨在全面理解和促进个体、群体乃至整个社会的心理健康状态。这个模型通常包含以下几个关键组成部分：
>
>- 认知因素：涉及个体的思维模式、信念系统、认知偏差以及解决问题的能力。认知因素对心理健康有重要影响，因为它们影响个体如何解释和应对生活中的事件。
>- 情感因素：包括情绪调节、情感表达和情感体验。情感健康是心理健康的重要组成部分，涉及个体如何管理和表达自己的情感，以及如何从负面情绪中恢复。
>- 行为因素：涉及个体的行为模式、习惯和应对策略。这包括应对压力的技巧、社交技能以及自我效能感，即个体对自己能力的信心。
>- 社会环境：包括家庭、工作、社区和文化背景等外部因素，这些因素对个体的心理健康有着直接和间接的影响。
>- 生理健康：身体健康与心理健康紧密相关。良好的身体健康可以促进心理健康，反之亦然。
>- 心理韧性：指个体在面对逆境时的恢复力和适应能力。心理韧性强的人更能够从挑战中恢复，并从中学习和成长。
>- 预防和干预措施：心理健康大模型还包括预防心理问题和促进心理健康的策略，如心理教育、心理咨询、心理治疗和社会支持系统。
>- 评估和诊断工具：为了有效促进心理健康，需要有科学的工具来评估个体的心理状态，以及诊断可能存在的心理问题。

 为了使我们的EmoLLM心理大模型有更好的表达效果，我们必须要有高质量的数据集。为了达到这一目标，我们决定利用四种强大的人工智能大模型：文心一言、通义千问、讯飞星火和智谱AI来蒸馏生成对话数据。此外，我们还将增强数据集的认知深度，通过加入少量自我认知数据集来提高模型的泛化能力。项目在ChatGLM3-6B、qwen_7b、internlm2_7b上均使用过数据集进行训练，并将internlm2_7b在OpenXLab上部署。
 >具体的微调步骤，请跟随我们的[文档](https://github.com/aJupyter/EmoLLM/blob/main/xtuner_config/README.md)一起学习吧💡

EmoLLM 是一个能够支持 理解用户-支持用户-帮助用户 心理健康辅导链路的心理健康大模型，由 [InternLM2](https://github.com/InternLM/InternLM) 指令微调而来，欢迎大家star~⭐⭐

## OpenXLab线上使用
EmoLLM大模型上线部署OpenXLab后，我们可以直接使用进行心理咨询
![image](https://github.com/Smiling-Weeping-zhr/Git/assets/127636623/244d9fa5-7cc3-4540-8f5d-31c5da615074)

## 微调
我们使用了xtuner微调和Lora微调两种形式，但是我们主要是对internlm2进行xtuner微调，Lora微调仅供兴趣拓展
### XTuner微调
使用XTuner对模型进行微调，具体的python脚本文件可以详见[文档](https://github.com/aJupyter/EmoLLM/tree/main/xtuner_config),并且脚本文件中有详细注解。
### Lora微调
使用 Transformers 的 Trainer 进行微调，具体详细脚本文件可以参考[文档](https://github.com/aJupyter/EmoLLM/blob/main/xtuner_config/ChatGLM3-6b-ft.md)。
若有疑问也可以联系作者，欢迎大家纠错和提出ideas💡,一起参与项目开发

## 数据集
数据集使用四种功能强大的人工智能大模型来生成，现[数据集](https://github.com/aJupyter/EmoLLM/tree/main/datasets)已开源
```json
{
        "conversation": [
            {
                "system": "现在你是一个心理专家，我有一些心理问题，请你用专业的知识帮我解决。",
                "input": "医生，我最近总是感到很焦虑，尤其是在学业上。我有个特别崇拜的同学，他好像在各方面都比我优秀，我总觉得自己怎么努力也追不上他，这让我压力特别大。",
                "output": "你好，首先感谢你对我敞开心扉。你的这种情绪其实很常见，这是由于过度比较和自我期待过高引发的焦虑情绪。我们要认识到每个人都有自己的发展节奏和优势，与他人比较并不是衡量自身价值的唯一标准。你可以试着列出自己在学习和其他方面的优点，同时理解并接纳自己的不足。我们可以一步步来，先从调整自我认知开始。"
            },
            {
                "input": "是的，我知道应该理性看待，但就是忍不住会去比较。我甚至晚上会因为这个睡不着觉，总想着怎样才能像他那样出色。",
                "output": "了解你的情况后，我建议你在睡前尝试进行放松训练，比如深呼吸、冥想等方法，有助于改善睡眠质量。另外，设定实际可达的目标也是很重要的，避免给自己设定过于严苛的标准。你是否可以具体描述一下你在学业上的困扰，我们一起看看如何制定一个适合你的个人提升计划？"
            },
            {
                "input": "我在数学方面总是比他落后很多，我尝试了很多方法提高，但还是觉得力不从心。",
                "output": "对于数学学习困难的问题，我们可以探讨更具针对性的学习策略，如分解复杂问题为小块、寻找优质的学习资源或参加学习小组等。同时，也要建立对失败的正确认识，失败是成功的垫脚石，每一次错误都是我们进步的机会。接下来，我会推荐一些有效的学习技巧，并鼓励你定期与我分享进展，以便我们根据实际情况进行调整。\n\n医生：另外，请记住，寻求帮助也是一种能力，如果你感到持续的压力和焦虑影响了生活和学习，不妨考虑找学校的心理辅导老师或者进一步接受专业心理咨询。心理健康与学业成就同样重要，让我们一起关注并呵护它。"
            }
        ]
    }
```

## 特别鸣谢
- [Sanbu](https://github.com/sanbuphy)
- [上海人工智能实验室](https://www.shlab.org.cn/)
- [闻星大佬（小助手）](https://github.com/vansin)
## 人员
[aJupyter](https://github.com/aJupyter)@南开大学在读硕士、datawhale成员

[jujimeizuo](https://github.com/jujimeizuo)@江南大学硕士

[Smiling&amp;Weeping](https://github.com/Smiling-Weeping-zhr)@哈尔滨工业大学（威海）在读本科生

[Farewell](https://github.com/8baby8)@

**Comment by jujimeizuo** [comment]
还缺人吗，可以一起吗，同acmer

**Reply by aJupyter** [reply]
是5班的吗

**Reply by jujimeizuo** [reply]
不是，我是12班的，应该可以跨班级组队吧？

**Reply by aJupyter** [reply]
先加俺吧 vx  aJupyter

**Comment by MING-ZCH** [comment]
看好噢，加油！
